src/coprocessor/dag/expr/compare.rs

```diff
@@ -230,15 +241,51 @@ where
     Ok(None)
 }
 
+#[inline]
+fn next_escaped(chars: &mut Chars, escape: char) -> Option<char> {
+    chars.next().and_then(|c| if c == escape {
+        chars.next().or_else(|| Some(c))
```

Comment:
I think this is wrong. "\%" should match "%" exactly instead of any string.

Reply:
```sql
mysql> select 'hello' like '%';
+------------------+
| 'hello' like '%' |
+------------------+
|                1 |
+------------------+
1 row in set (0.00 sec)
```
It comes from MySQL.

Reply:
Github ate my "\\", what I mean is "\\%".